### PR TITLE
Implement basic structure for entity class

### PR DESCRIPTION
**Pull request by ivantjh**
@chesterhow @JxJoker Help me check if this makes sense. I will make player class inherit from this and move all the logic to player class.

### CODE REVIEW

File: src/threechances.h [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
 #include "textureManager.h"
 #include "image.h"
+#include "entity.h"
 
 class ThreeChances : public Game {
 private:
 	// Game items
 	TextureManager mapTexture;
 	Image		   map;
+	Entity		   bob;
```
Comment (chesterhow) [inline src/threechances.h]: No need to use Entity here. Just use the respective classes later

**Review by chesterhow — CHANGES_REQUESTED** [review]

File: src/entity.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	// setters
+	void setHealth(float h) { health = h; }
+	void setDamage(float d) { damage = d; }
+	void setAtkRange(int atkR) { atkRange = atkR; }
+	void setMoveRange(int mr) { moveRange = mr; }
+	void setLevel(std::vector<int> l) { levels = l; }
+	void setAbilities(std::vector<entityNS::ABILITY_TYPE> a) { abilities = a; }
+
+	// can be overrode
```
Comment (chesterhow) [inline src/entity.h]: overridden...

File: src/entity.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	enum ABILITY_TYPE { TRAVEL_WATER, TRAVEL_LAVA, TRAVEL_WALL, BURNS_PLAYER, SHOOTS_FIRE };
+}
+
+struct EntityData {
+	float health;
+	float damage;
+	int atkRange;
+	int moveRange;
+	std::vector<int> levels;
+	std::vector<entityNS::ABILITY_TYPE> abilities;
```
Comment (chesterhow) [inline src/entity.h]: Add a variable for moves?

**Review by chesterhow — CHANGES_REQUESTED** [review]

File: src/entity.h (hunk 3 above)
Comment (ivantjh) [reply src/entity.h]: oh ya, will do this later

**Review by chesterhow — APPROVED** [review]

**Pull request merged**